src/mkdocstrings/_cache.py

```diff
@@ -4,20 +4,26 @@
 import os
 import urllib.parse
 import urllib.request
-from typing import BinaryIO, Callable
+from typing import BinaryIO, Callable, Union
 
 import click
 import platformdirs
+import regex as re
 
 from mkdocstrings.loggers import get_logger
 
 log = get_logger(__name__)
 
+# Regex pattern for an environment variable
+ENV_VAR_PATTERN = re.compile(r"^\$([A-Za-z_][A-Za-z0-9_]*)$")
```

Comment:
I think I'd prefer directly going with `${...}` syntax, to clearly delimit the variable names. Otherwise I fear it will be asked later and we'll have to support both :sweat_smile: WDYT? It's also [what PDM does](https://pdm-project.org/en/latest/usage/dependency/#hide-credentials-in-the-url), for example.

Reply:
So only `${...}`, or both? I personally prefer `${...}` as well, but felt the pattern became a bit too complicated when trying to support both:
```python
ENV_VAR_PATTERN = re.compile(r"^\$([A-Za-z_][A-Za-z0-9_]*)$|^\$\{([A-Za-z_][A-Za-z0-9_]*)\}$")
```
plus having to modify the `match.group(1)` to now check and extract from both capture groups.

Reply:
Only `${...}` :slightly_smiling_face:

Reply:
I've changed it to `${...}` in [`3c8a294` (#710)](https://github.com/mkdocstrings/mkdocstrings/pull/710/commits/3c8a2946d15b3c8fd00706d77bb44bdccbfadf2a)